### PR TITLE
allow streams api to be filtered like list api

### PR DESCRIPTION
**Pull request by ripienaar**
Signed-off-by: R.I.Pienaar <rip@devco.net>

/cc @nats-io/core

### CODE REVIEW

File: server/jetstream_cluster_test.go [hunk 1]
```diff
@@ -9442,6 +9442,52 @@ func TestJetStreamClusterAccountInfoForSystemAccount(t *testing.T) {
 	}
 }
 
+func TestJetClusterStreamListFilter(t *testing.T) {
```
Comment (derekcollison) [inline server/jetstream_cluster_test.go:9445]: Can be made into one test that tests single server and cluster, I have a few examples of that.

File: server/jetstream_cluster.go [hunk 2]
```diff
@@ -3848,8 +3848,29 @@ func (s *Server) jsClusteredStreamListRequest(acc *Account, ci *ClientInfo, offs
...
 	for _, sa := range cc.streams[acc.Name] {
-		streams = append(streams, sa)
+		if IsNatsErr(sa.err, JSClusterNotAssignedErr) {
+			continue
+		}
+
+		if filter != _EMPTY_ {
+			// These could not have subjects auto-filled in since they are raw and unprocessed.
+			if len(sa.Config.Subjects) == 0 {
+				if SubjectsCollide(filter, sa.Config.Name) {
```
Comment (derekcollison) [inline server/jetstream_cluster.go:3858]: Should it be collide or do we want the filter to be equal or a subset of the subjects attached to the stream?

File: server/jetstream_cluster_test.go (hunk 1 above)
Comment (ripienaar) [reply server/jetstream_cluster_test.go:9445]: Ok

File: server/jetstream_cluster.go (hunk 2 above)
Comment (ripienaar) [reply server/jetstream_cluster.go:3858]: Don’t know tbh. I copied logic from stream names they should behave the same. So this is whatever names does :)

Comment (derekcollison) [reply server/jetstream_cluster.go:3858]: ok so this is from names filter code, so agree using same thing makes sense here.

**Comment by ripienaar** [comment]
Updated to one test

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**